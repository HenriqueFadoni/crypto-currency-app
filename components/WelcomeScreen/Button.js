import React from 'react';
import { Button } from 'react-native';

const ButtonDefault = props => (
    <Button
        title="Search Currency"
        onPress={() => {
            props.navigation.navigate({ routeName: 'currencyList' });
        }}
    />
);
export default ButtonDefault;