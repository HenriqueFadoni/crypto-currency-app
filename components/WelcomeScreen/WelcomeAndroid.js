import React, { Component } from 'react';
import {
    StyleSheet,
    View,
    Text,
    Animated,
} from 'react-native';

import ButtonDefault from './Button';

const WelcomeAndroid = props => (
    <>
        <Animated.View style={[styles.titlesContainerAndroid, styles.animation]}>
            <Text style={styles.subTitle}>{props.title}</Text>
            <Text style={styles.title}>{props.subTitle}</Text>
        </Animated.View>
        <View style={styles.btnContainer}>
            <ButtonDefault color='blue' navigation={props.navigation} />
        </View>
    </>
);

const styles = StyleSheet.create({
    titlesContainerAndroid: {
        borderRadius: 1000,
        height: 250,
        width: 250,

        alignSelf: 'center',
        justifyContent: 'center',
        paddingTop: 100,
        marginBottom: 100,

        borderStyle: 'solid',
        borderColor: 'black',
        borderWidth: 2,
    },
    title: {
        textAlign: 'center',
        fontSize: 25,
        marginBottom: 100
    },
    subTitle: {
        textAlign: 'center',
        fontSize: 20,
    },
    btnContainer: {
        width: 100,
        justifyContent: "center",
        alignSelf: "center"
    }
});

export default WelcomeAndroid;