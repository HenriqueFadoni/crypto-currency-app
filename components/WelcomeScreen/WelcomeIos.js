import React, { Component } from 'react';
import {
    StyleSheet,
    Text,
    Animated,
} from 'react-native';

import ButtonDefault from './Button';

const WelcomeIos = props => (
    <>
        <Animated.View style={[styles.titlesContainerIos, props.animation]}>
            <Text style={styles.subTitle}>{props.title}</Text>
            <Text style={styles.title}>{props.subTitle}</Text>
        </Animated.View>
        <ButtonDefault navigation={props.navigation} />
    </>
);

const styles = StyleSheet.create({
    titlesContainerIos: {
        borderRadius: 1000,
        height: 300,
        width: 300,
        alignSelf: 'center',
        paddingTop: 120,
        marginBottom: 100,
        borderStyle: 'solid',
        borderColor: 'black',
        borderWidth: 2,
    },
    title: {
        textAlign: 'center',
        fontSize: 25,
        marginBottom: 100
    },
    subTitle: {
        textAlign: 'center',
        fontSize: 20,
    }
});

export default WelcomeIos;