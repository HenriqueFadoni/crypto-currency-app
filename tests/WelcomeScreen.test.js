import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';

// Configuring Adapter
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

// Importing File
import WelcomeScreen from '../screens/WelcomeScreen';
import ButtonDefault from '../components/Button';

describe('WelcomeScreen', () => {
    //Simple Unit Testing
    it('Checks if WelcomeScreen rendered 2 children', () => {
        const WScreen = shallow(<WelcomeScreen />);
        expect(WScreen.children.length).toBe(2);
    });

    it('WelcomeScreen should have ButtonDefault as child', () => {
        const wrapper = shallow(<WelcomeScreen />).setProps({
            navProps: 'currencyList'
        });
        expect(wrapper.find(ButtonDefault)).toHaveLength(1);
    });

    //Snapshot Testing
    it('Checks if everything is being rendered accordingly', () => {
        const tree = renderer.create(<WelcomeScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});