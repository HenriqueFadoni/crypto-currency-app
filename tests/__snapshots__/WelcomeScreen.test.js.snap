// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WelcomeScreen Checks if everything is being rendered accordingly 1`] = `
<View
  style={
    Object {
      "alignContent": "center",
      "height": "100%",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "opacity": 0,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      WELCOME TO
    </Text>
    <Text
      style={
        Object {
          "fontSize": 25,
          "marginBottom": 100,
          "textAlign": "center",
        }
      }
    >
      CRYPTO-WATCH
    </Text>
  </View>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Search Currency
      </Text>
    </View>
  </View>
</View>
`;
